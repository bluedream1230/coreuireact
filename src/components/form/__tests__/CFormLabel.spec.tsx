import * as React from 'react'
import { render /* ,screen */ /*,fireEvent */ /* ,waitFor */ } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { CFormLabel } from '../../../index'

test('loads and displays CFormLabel component', async () => {
  const { container } = render(<CFormLabel>Test</CFormLabel>)
  expect(container).toMatchSnapshot()
})

test('CFormLabel customize className', async () => {
  const { container } = render(<CFormLabel className="bazinga">Test</CFormLabel>)
  expect(container).toMatchSnapshot()
  expect(container.firstChild).toHaveClass('bazinga')
  expect(container.firstChild).toHaveClass('form-label')
})

test('CFormLabel customize className and classNameParent', async () => {
  const { container } = render(
    <CFormLabel className="bazinga" classNameParent="classNameParent">
      Test
    </CFormLabel>,
  )
  expect(container).toMatchSnapshot()
})
