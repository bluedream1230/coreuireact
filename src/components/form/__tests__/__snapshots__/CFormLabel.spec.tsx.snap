// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CFormLabel customize className 1`] = `
<div>
  <label
    class="form-label bazinga"
  >
    Test
  </label>
</div>
`;

exports[`CFormLabel customize className and classNameParent 1`] = `
<div>
  <label
    class="classNameParent"
  >
    Test
  </label>
</div>
`;

exports[`loads and displays CFormLabel component 1`] = `
<div>
  <label
    class="form-label"
  >
    Test
  </label>
</div>
`;
